### DISCUSSION TITLE
How can i use middleware for specific handler only?

### DISCUSSION THREAD

**Discussion by fineshop** [honojs/hono#1645] [Q&A]
```ts
app.get(
  "*",
  cache({...}),
  serveStatic({...}),
  async (c, next) => {
    return new Response("Hello");
  }
);
```

How can i use the `cache({...})` only for `serveStatic({...})` and not for other handlers?

**Comment by yusukebe** [comment]
What about write the following?

```ts
app.get('*', cache({ cacheName: 'foo' }), serveStatic())
```

**Comment by fineshop** [comment]
> What about write the following?
> 
> ```ts
> app.get('*', cache({ cacheName: 'foo' }), serveStatic())
> ```

I tried the following:

```ts
app.get("*", cache({...}), serveStatic({...}));

app.get("*", async () => new Response("Hello"));
```

But the cache headers are being set to both middleware and handler.
I want to cache only the static assets and the next handlers without being cached.

**Reply by fineshop** [reply]
Any solution or workaround?

**Reply by yusukebe** [reply]
@fineshop 

How about this?

```ts
app.get('*', async (c, next) => {
  await cache({ cacheName: 'foo' })(c, next).then(() => serveStatic()(c, next))
})
```

**Reply by fineshop** [reply]
It throws an error:

```
Error: next() called multiple times
```

**Comment by yusukebe** [comment]
Hi @fineshop 

I've thought this issue but, it's seems to be impossible. The handlers and middleware are applied for the path such as `*`. So if you want to use middleware according to conditions, you have to specify them with a path definition like using a directory name or an extension:

```ts
app.get('/static/*', cache(), serveStatic())
```

Or

```ts
app.get('/:filename{.+\\.txt$}', cache(), serveStatic())
```
